### ISSUE TITLE
Some nodes can only use fixed values

### ISSUE THREAD

**Issue by CodeLazier** [limbonaut/limboai#224]
### Problem statement

Can built-in decorator nodes only use fixed values? For example, the Duration of Cooldown and the Run Limit of RunLimit can be changed to refer to BBNode, similar to the animation playback node of Scene, which greatly increases flexibility.Even all the properties can be set directly or by code using BBNode.

### Proposed solution

none

### Alternatives

none

**Comment by limbonaut** [comment]
`BBNode` is not designed to reference an object property. It's strictly for referencing nodes. It's also not a recommended approach to reference nodes directly from a `BehaviorTree`.

**Comment by CodeLazier** [comment]
So what are the best practices for those who need runtime changes? Can I only use BTask to implement the functions that the decorator already has?

**Comment by CodeLazier** [comment]
I've found that it's possible to define the value of the setting yourself by inheriting from the decorator class.

**Comment by limbonaut** [comment]
Ah, you meant BBParam. I tried to use BBParam only where it's most needed, because it takes more clicks to edit parameters in GDExtension (unlike in module version). If this PR gets accepted, we can adopt it more widely: 
- https://github.com/godotengine/godot/pull/87375

**Comment by CodeLazier** [comment]
Extending BTDecorator does what I expected so far, but Limboai demo doesn't seem to have an example of this, I thought it was in BTDecorator
If a SUCCESS is returned in the _tick method, Limbo will automatically execute its child node. But tested it wasn't the case, I needed to call it manually.
For example, if I want to implement my own defined BTColldown decorator, then:

```gdscript
extends BTCooldown

func _setup() -> void:
	duration = agent.attack_frequency


func _tick(delta: float) -> Status:
	return get_child(0).execute(delta)

```
At the moment this code works, so is this the correct usage?

**Comment by limbonaut** [comment]
> For example, if I want to implement my own defined BTColldown decorator, then
That's not how inheritance work. If you override `_tick` implementation, you override it fully, so cooldown functionality is no longer executed, and it is replaced by your own functionality, which is simply executing a child and returning its status. Have you tried not overriding `_tick` in that example?

**Comment by CodeLazier** [comment]
If I don't override _tick method, then Cooldown will never execute the child node.
```gdscript
extends BTCooldown

func _setup() -> void:
	duration = agent.attack_frequency
```


If I don't cover anything, including _setup, that custom Cooldown works.
```gdscript
extends BTCooldown

```

**Comment by limbonaut** [comment]
Right, that may be a bug. Since you are overriding `_setup` it prevents `BTCooldown` from executing its own setup routine. It shouldn't override the existing setup behavior, I think, since setup is like a constructor in a sense.

**Comment by limbonaut** [comment]
> Right, that may be a bug. Since you are overriding `_setup` it prevents `BTCooldown` from executing its own setup routine. It shouldn't override the existing setup behavior, I think, since setup is like a constructor in a sense.

Fixing it in #227.
